File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map=new Map();

    for(let i=0;i<nums.length;i++){
       let diff=target-nums[i];

       if(map.has(diff)){
        return [map.get(diff),i];
        
       }
       map.set(nums[i],i);
    }

};

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> seen;
        
        for(int i=0;i<nums.size();i++){
            int need=target-nums[i];
            auto it=seen.find(need);
            if(it!=seen.end()) return {it->second,i};
            seen[nums[i]]=i;
        }
        

        return {};
    }
};